src/terms/nonlocal.jl

```diff
@@ -227,3 +227,85 @@ function build_form_factors(psp, qs::Array)
     end
     form_factors
 end
+
```

Comment:
These changes are again a lot of code duplication, which can be avoided with the tricks mentioned above.